### PR TITLE
Android Embedding PR22: Polish - FlutterActivity Intent factories, FlutterFragment control of render modes, FlutterSurfaceView transparent until rendering is ready.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Android Embedding PR22: Polish - FlutterActivity Intent factories, FlutterFragment control of render modes, FlutterSurfaceView transparent until rendering is ready.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java [hunk 1]
```diff
@@ -90,20 +93,26 @@ public static FlutterFragment newInstance() {
    * @param appBundlePath the path to the app bundle which contains the Dart app to execute, defaults
    *                      to {@link FlutterMain#findAppBundlePath(Context)}
    * @param flutterShellArgs any special configuration arguments for the Flutter engine
+   * @param renderMode render Flutter either as a {@link FlutterView.RenderMode#surface} or a
+   *                   {@link FlutterView.RenderMode#texture}. Use {@code texture} for situations
+   *                   where this {@code FlutterFragment} might have content on top of it, such as
+   *                   a drawer.
```
Comment (mklim) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: nit: It's hard to get into the details a ton in just a line or two here but as is this isn't letting people know that using `texture` most likely comes with a huge performance cost.  I'd probably rephrase to something along the lines of "Use `{@code texture}` when you're willing and able to sacrifice performance to have this `{@code FlutterFragment}` interleave with other Android views. `{@code surface}` is a much more performant default, but won't composite with complicated View hierarchies." Exact wording doesn't matter much to me but I think a really rough summary of the pros/cons would be worth it here.

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java [hunk 2]
```diff
@@ -326,6 +341,18 @@ protected String getDartEntrypointFunctionName() {
...
 
+  /**
+   * Returns the desired {@link FlutterView.RenderMode} for the {@link FlutterView} displayed in
+   * this {@code FlutterFragment}.
+   *
+   * Defaults to {@link FlutterView.RenderMode#surface}.
+   */
+  @NonNull
+  protected FlutterView.RenderMode getRenderMode() {
+    String renderModeName = getArguments().getString(ARG_FLUTTERVIEW_RENDER_MODE, null);
```
Comment (mklim) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: subjective, nit: If you set the default value here to `FlutterView.RenderMode.surface.toString()` you could have the next statement consistently return without needing a null check and ternary.

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java (hunk 1 above)
Comment (matthew-carroll) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: done

**Comment by matthew-carroll** [comment]
@jason-simmons @mklim @mjohnsullivan @tvolkert I just pushed a commit that converts `FlutterActivity` and `FlutterFragment` factories to builders for brevity and future expansion options.

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java (hunk 2 above)
Comment (matthew-carroll) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: done

**Pull request merged**